2-projects/1-todo-list/script.js

```diff
@@ -13,13 +45,48 @@ let todos = [
 populateTodoList(todos);
 
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
+const addBtn = document.getElementsByClassName("btn btn-primary mb-3")[0];
+addBtn.addEventListener("click", addNewTodo);
+
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  // Write your code here... and remember to reset the input field to be blank aft er creating a todo!
+  let inputElement = document.getElementById("todoInput");
+  if (inputElement.value === "") {
+    
+  } else {
+    let ulFlusher = document.getElementById("todo-list");
+    ulFlusher.innerHTML = "";
+    let newItem = {};
+    newItem["task"] = inputElement.value;
+    newItem["completed"] = false;
+    todos.push(newItem);
+    populateTodoList(todos);
+  }
```

Comment:
Here the initial `if` statement's scope doesn't hold any commands, and you do the conditional computation in the `else` branch.

You could opt for `if (inputElement.value !== "")`, put all your statements there, and then you don't need the else branch.

like:
```
if (inputElement.value !== "") {
let ulFlusher = document.getElementById("todo-list");
    ulFlusher.innerHTML = "";
    let newItem = {};
    newItem["task"] = inputElement.value;
    newItem["completed"] = false;
    todos.push(newItem);
    populateTodoList(todos);
  }
```